pkg/performanceprofile/controller/performanceprofile_controller_hypershift.go

```diff
@@ -71,3 +102,230 @@ func (r *PerformanceProfileReconciler) hypershiftSetupWithManager(mgr ctrl.Manag
+	if err != nil {
+		klog.Warningf("failed to parseTunedManifests in ConfigMap %s: %v", instance.ObjectMeta.Name, err)
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	pinningMode, err := r.getInfraPartitioningMode()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	//REVIEW - jlom Not fully sure if `getContinerRuntimeName` gonna work in hypershift env
+	ctrRuntime, err := r.getContainerRuntimeName(ctx, performanceProfileFromConfigMap)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("hypershift could not determine high-performance runtime class container-runtime for profile %q; %w", performanceProfileFromConfigMap.Name, err)
+	}
+	klog.Infof("hypershift using %q as high-performance runtime class container-runtime for profile %q", ctrRuntime, performanceProfileFromConfigMap.Name)
+
+	componentSet, err := manifestset.GetNewComponents(performanceProfileFromConfigMap, nil, &pinningMode, ctrRuntime)
+	if err != nil {
+		klog.Errorf("failed to deploy performance profile from configMap %q components: %v", instance.Name, err)
+		//REVIEW - Should we record this events? and if so where? in the CM or the PP?
+		return reconcile.Result{}, err
+	}
+
+	//jlom - Now we have to create a ConfigMap for each of the elements in the componentSet and then handle them to
+	//       the different agents that would made them effective in the managed cluster.( where workers are)
+	tunedEncoded, err := encodeTuned(componentSet.Tuned)
+	if err != nil {
+		klog.Warningf("failed to Encode Tuned in ConfigMap %s: %v", instance.ObjectMeta.Name, err)
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	tunedConfigMap := TunedConfigMap(instance, performanceProfileFromConfigMap.Name, cmNodePoolNamespacedName, string(tunedEncoded))
+
+	if err := createOrUpdateTunedConfigMap(tunedConfigMap, ctx, r.Client); err != nil {
+		klog.Error("failure on Tuned process: %v", err)
+		// Return and don't requeue
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func encodeTuned(tuned *tunedv1.Tuned) ([]byte, error) {
+	scheme := runtime.NewScheme()
+	tunedv1.AddToScheme(scheme)
+	tunedEncoded, err := encodeManifest(tuned, scheme)
+	return tunedEncoded, err
+}
+
+func encodeManifest(obj runtime.Object, scheme *runtime.Scheme) ([]byte, error) {
+	yamlSerializer := serializer.NewSerializerWithOptions(
+		serializer.DefaultMetaFactory, scheme, scheme,
+		serializer.SerializerOptions{Yaml: true, Pretty: true, Strict: true},
+	)
+
+	buff := bytes.Buffer{}
+	err := yamlSerializer.Encode(obj, &buff)
+	return buff.Bytes(), err
+}
+
+func createOrUpdateTunedConfigMap(cm *corev1.ConfigMap, ctx context.Context, cli client.Client) error {
+	tunedConfigMapUpdateFunc := func(orig, dst *corev1.ConfigMap) error {
+		//REVIEW - Maybe here I should ensure the readed ConfifMap has the needed labels and annotations.
```

Comment:
seems like a good idea, yes. Can be done as followup though.

Reply:
issue open.
comment deleted